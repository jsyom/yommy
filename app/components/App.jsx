import React, { Component } from 'react';
import Header from './Header';
import Jumbo from './Jumbo';
import About from './About';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: ''
    };
  }

  render() {
    return (
      <div className="app-container">
        <Header />
        <Jumbo />
        <About />
      </div>
    );
  }
}
