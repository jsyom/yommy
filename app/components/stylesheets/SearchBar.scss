@import 'Media.scss';

.search {
  width: 100%;
  position: relative;
}

.search-term {
  font-family: 'Source Sans Pro', sans-serif;
  float: left;
  width: 100%;
  border: 3px solid #355C96;
  padding: 5px;
  height: 20px;
  border-radius: 5px;
  outline: none;
  color: #355C96;
  font-size: 15px;
    @include respond-to(handhelds) {
      width: 90%;
    }
}

.search-term:focus{
  color: #080808;
}

.search-button {
  position: absolute;
  right: -50px;
  width: 45px;
  height: 36px;
  border: 1px solid #355C96;
  background: #355C96;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  cursor: pointer;
  font-size: 20px;
    @include respond-to(handhelds) {
      right: 0px;
    }
}

.search-container {
  width: 80%;
  margin-top: 1%;
  margin-left: 1%;
}
