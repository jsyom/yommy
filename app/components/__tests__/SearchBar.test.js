import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchBar from '../SearchBar';
import App from '../App';

test('SearchBar component should render as expected', () => {
	const component = shallow(<SearchBar />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});

test('onInputChange updates the title when typed into the input', () => {
	const component = shallow(<SearchBar />);
	component.find('input').simulate('change', { target: { value: 'superman' } });
	expect(component.state().title).toEqual('superman');
	expect(toJson(component)).toMatchSnapshot();
});

test('component should receive props', () => {
	const formSubmit = jest.fn();
	const component = mount(<SearchBar formSubmit={formSubmit} />);
	expect(component.props().formSubmit).toBeDefined();
	expect(toJson(component)).toMatchSnapshot();
});


