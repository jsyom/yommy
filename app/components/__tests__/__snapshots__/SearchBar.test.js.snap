// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should render as expected 1`] = `
<div
  className="search-container"
>
  <form
    className="search"
    onSubmit={[Function]}
  >
    <input
      className="search-term"
      onChange={[Function]}
      placeholder="Enter a movie"
      value=""
    />
    <button
      className="search-button"
      type="submit"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </form>
</div>
`;

exports[`component should receive props 1`] = `
<SearchBar
  formSubmit={[Function]}
>
  <div
    className="search-container"
  >
    <form
      className="search"
      onSubmit={[Function]}
    >
      <input
        className="search-term"
        onChange={[Function]}
        placeholder="Enter a movie"
        value=""
      />
      <button
        className="search-button"
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>
</SearchBar>
`;

exports[`onInputChange updates the title when typed into the input 1`] = `
<div
  className="search-container"
>
  <form
    className="search"
    onSubmit={[Function]}
  >
    <input
      className="search-term"
      onChange={[Function]}
      placeholder="Enter a movie"
      value="superman"
    />
    <button
      className="search-button"
      type="submit"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </form>
</div>
`;
